src/Simplify.cpp

```diff
@@ -4274,6 +4349,16 @@ void check_algebra() {
     check((x - y)*-2, (y - x)*2);
     check((xf - yf)*-2.0f, (yf - xf)*2.0f);
 
+    // Pull terms that are a multiple of the divisor out of a ternary expression
+    check(((x*4 + y) + z) / 2, x*2 + (y + z)/2);
+    check(((x*4 - y) + z) / 2, x*2 - (y - z)/2);
```

Comment:
Oh yeah, I see one broken existing rule. Will fix in master with
parentheses fiddling as you suggest.

On Thu, Jul 7, 2016 at 3:36 PM, Jing Pu <notifications@github.com> wrote:

> In src/Simplify.cpp
> <https://github.com/halide/Halide/pull/1349#discussion_r70000242>:
>
> > @@ -4274,6 +4349,16 @@ void check_algebra() {
> >      check((x - y)*-2, (y - x)*2);
> >      check((xf - yf)*-2.0f, (yf - xf)*2.0f);
> >
> > +    // Pull terms that are a multiple of the divisor out of a ternary expression
> > +    check(((x*4 + y) + z) / 2, x*2 + (y + z)/2);
> > +    check(((x*4 - y) + z) / 2, x*2 - (y - z)/2);
>
> I see your point. Then the previous rules that simplify divisions of
> binary expressions would be wrong in the same way. For example, check((x*4
> - y)/2, x*2 - y/2);. Can we fix it by rearranging the parenthesis?
> Something like:
>
> check((x*4 - y)/2, x*2  + (- y)/2);
> check(((x*4 - y) + z) / 2, x*2 + (- y + z)/2);
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/halide/Halide/pull/1349/files/ab4e0af24fbe612ec516374866d134238c557d6b#r70000242>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAfdRmfEYKE0aAWQZeCrPTrpb6UDBR5gks5qTX-EgaJpZM4JBakb>
> .
>